vsphere/datadog_checks/vsphere/utils.py

```diff
@@ -51,40 +58,66 @@ def match_any_regex(string, regexes):
     return False
 
 
-def is_resource_excluded_by_filters(mor, infrastructure_data, resource_filters):
-    # type: (vim.ManagedEntity, InfrastructureData, ResourceFilters) -> bool
+def is_resource_collected_by_filters(mor, infrastructure_data, resource_filters, tags_cache):
+    # type: (vim.ManagedEntity, InfrastructureData, ResourceFilters, TagsCache) -> bool
     resource_type = MOR_TYPE_AS_STRING[type(mor)]
 
-    if not [f for f in resource_filters if f[0] == resource_type]:
-        # No filter for this resource, collect everything
+    if not [f for f in resource_filters if f.resource == resource_type and f.is_whitelist]:
+        # No whitelist filter specified for this resource, consider that the resource matches the whitelist
+        match_whitelist = True
+    else:
+        match_whitelist = _does_resource_match_filters(
+            mor, infrastructure_data, resource_filters, tags_cache, is_whitelist=True
```

Comment:
It might be cleaner to have two lists of resource filters, one for blacklist and one for whitelist instead of using booleans.